src/services/completions.ts

```diff
@@ -778,7 +779,11 @@ namespace ts.Completions {
             if (!entry) {
                 return undefined;
             }
-            ({ insertText, isSnippet, importAdder, sourceDisplay } = entry);
+            ({ insertText, isSnippet, importAdder, labelDetails } = entry);
+            if (!preferences.includeCompletionsWithLabelDetails) {
+                name = name + labelDetails.detail;
+                labelDetails = undefined;
+            }

```

Comment:
This is interesting - so the idea here is to always include the completions with label detials, but modify the `name` property. As an API consumer, I would assume that this flag (like the other similarly named options) implies that these completions don't show up at all, not that the object format changes.

Maybe `useLabelDetailsInCompletionEntries` or something like that?